packages/react-strict-dom/src/native/modules/ContextInheritedStyles.js

```diff
@@ -34,7 +34,17 @@ export function ProvideInheritedStyles(
 ): React$MixedElement {
   const { children, value } = props;
   const inheritedStyles = useInheritedStyles();
-  const flatStyle = flattenStyle([inheritedStyles as ?Style, value as ?Style]);
+  let flatStyle;
+  if (
+    value == null ||
+    (typeof value === 'object' && Object.keys(value).length === 0)
+  ) {
+    flatStyle = inheritedStyles;
+  } else if (inheritedStyles === defaultContext) {
+    flatStyle = value;
+  } else {
+    flatStyle = flattenStyle([inheritedStyles as ?Style, value as ?Style]);
+  }
```

Comment:
Let's make this always return the same object if the result hasn't changed. We can borrow Relay's recycleNodesInto() utility and do the following:

```js
const [flattenedStyles, setFlattenedStyles] = useState(null);
const nextFlattenedStyles = recycleNodesInto(flattenedStyles, flattenStyle(inheritedStyles, value));
if (nextFlattenedStyles !== flattenedStyles) {
  setFlattenedStyles(nextFlattenedStyles);
}
return <ContextInheritedStyles.Provider value={flattenedStyles} />
```

Reply:
actually, it's probably also worth wrapping the `recycleNodes(..., flattenStyle(...))` bit in a useMemo just to save re-evaluating if the component happens to re-render